src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -1638,6 +1638,26 @@ public static Conversion ClassifyConversion(this SemanticModel? semanticModel, i
             var csModel = semanticModel as CSharpSemanticModel;
             return csModel?.GetInterceptorMethod(node, cancellationToken);
         }
+
+        /// <summary>
+        /// If 'node' cannot be intercepted syntactically, returns null.
+        /// Otherwise, returns an instance which can be used to intercept the call denoted by <paramref name="node"/>.
+        /// </summary>
+        [Experimental(RoslynExperiments.Interceptors, UrlFormat = RoslynExperiments.Interceptors_Url)]
+        public static InterceptableLocation? GetInterceptableLocation(this SemanticModel? semanticModel, InvocationExpressionSyntax node, CancellationToken cancellationToken = default)
+        {
+            var csModel = semanticModel as CSharpSemanticModel;
+            return csModel?.GetInterceptableLocation(node, cancellationToken);
+        }
+
+        /// <summary>
+        /// Gets an attribute list syntax consisting of an InterceptsLocationAttribute, which intercepts the call referenced by parameter <paramref name="location"/>.
+        /// </summary>
+        [Experimental(RoslynExperiments.Interceptors, UrlFormat = RoslynExperiments.Interceptors_Url)]
+        public static string GetInterceptsLocationAttributeSyntax(this InterceptableLocation location)
+        {
+            return $"""[global::System.Runtime.CompilerServices.InterceptsLocationAttribute({location.Version}, "{location.Data}")]""";
```

Comment:
>location.Version [](http://example.com/codeflow?start=93&length=16)

Should we validate that `location.Version` is positive?
Should we validate that `location.Data` is not null?
Can we check how version numbers get output for a range of version numbers? (I think it should behave correctly already, but consider testing with version number of than 1)